integration/deterministic-build-output-test.ts

```diff
@@ -3,115 +3,204 @@ import globby from "globby";
+          });
+        `,
+        "app/styles/side-effect.css": css`
+          .side-effect {
+            color: mintcream;
+          }
+        `,
+      },
+    },
+  },
+  {
+    name: "Vanilla Extract with cache enabled",
+    init: {
+      future: {
+        unstable_cssModules: true,
+        unstable_cssSideEffectImports: true,
+        unstable_postcss: true,
+        unstable_vanillaExtract: { cache: true },
+        v2_routeConvention: true,
+      },
+      files: {
+        "app/routes/_index.mdx": "# hello world",
+        "app/routes/foo.tsx": js`
+          export * from "~/foo/bar.server";
+          import styles from "~/styles/foo.module.css";
+          import { vanilla } from "~/styles/vanilla.css";
+          import "~/styles/side-effect.css";
+          export default () => <div className={[styles.foo, vanilla].join(' ')}>YAY</div>;
+        `,
+        "app/foo/bar.server.ts": "export const meta = () => []",
+        "app/styles/foo.module.css": css`
+          .foo {
+            background-image: url(~/images/foo.svg);
+            composes: bar from "~/styles/bar.module.css";
+            composes: baz from "./baz.module.css";
+          }
+        `,
+        "app/styles/bar.module.css": css`
+          .bar {
+            background-color: peachpuff;
           }
-        ]);
-      `,
-      "app/styles/chocolate.css.ts": css`
-        import { style } from "@vanilla-extract/css";
+        `,
+        "app/styles/baz.module.css": css`
+          .baz {
+            color: coral;
+          }
+        `,
+        "app/images/foo.svg": `
+          <svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
+            <circle cx="50" cy="50" r="50" fill="coral" />
+          </svg>
+        `,
+        "app/styles/vanilla.css.ts": css`
+          import { style } from "@vanilla-extract/css";
+          import { chocolate } from "./chocolate.css";
+          import imageUrl from "~/images/foo.svg";
 
-        export const chocolate = style({
-          color: "chocolate",
-        });
-      `,
-      "app/styles/side-effect.css": css`
-        .side-effect {
-          color: mintcream;
-        }
-      `,
+          export const vanilla = style([
+            chocolate,
+            {
+              backgroundImage: [
+                "url(" + imageUrl + ")",
+                "url(~/images/foo.svg)",
+              ],
+            }
+          ]);
+        `,
+        "app/styles/chocolate.css.ts": css`
+          import { style } from "@vanilla-extract/css";
+
+          export const chocolate = style({
+            color: "chocolate",
+          });
+        `,
+        "app/styles/side-effect.css": css`
+          .side-effect {
+            color: mintcream;
+          }
+        `,
+      },
     },
-  };
-  let dir1 = await createFixtureProject(init);
-  let dir2 = await createFixtureProject(init);
+  },
+];
 
-  expect(dir1).not.toEqual(dir2);
+testCases.forEach((testCase) => {
```

Comment:
Grouping tests in loops like this makes me a bit uneasy. Makes it tougher to isolate failures IMO. I don't mind using the same files config for each test, but do you think we could run each actual test explicitly instead?